src/utils/Cache.ts

```diff
@@ -99,29 +139,85 @@ export class Cache {
    * @param key - The key used to identify the cached value
    * @returns value - if the cache contains the key
    */
-  static has(key: string) {
+  static async has(key: string) {
     try {
+      const tmp = await Cache.get(key);
       const expiration = Date.now() - 86400000;
-      const tmp = storageReference.getItem(key);
-      if (tmp !== undefined && tmp && tmp !== '') {
-        const val = JSON.parse(tmp);
-        if (val.dateCached) {
-          return val.dateCached > expiration;
+        if (tmp !== undefined && tmp && tmp !== '') {
+          if (tmp.dateCached) {
+            return tmp.dateCached > expiration;
+          }
+          return Promise.resolve(true);
         }
-
-        return true;
-      }
-      return false;
+        return Promise.resolve(false);
     } catch (err) {
       console.warn(`An error occurred while looking for ${key} in Cache`, err);
-      return false;
+      return Promise.resolve(false);
     }
   }
   /**
    * Removes a key from the cache
    * @param key - The key used to identify the cached value
    */
-  static remove(key: string) {
+  static async remove(key: string) {
+    if (localforage !== undefined) {
+      await localforage.ready();
+      await localforage.removeItem(key);
+      return;
+    }
+
     storageReference.removeItem(key);
   }
+
+  /**
+   * Clears out the cache
+   * @param key - The key used to identify the cached value
+   */
+  static async clear() {
+    if (localforage !== undefined) {
+      await localforage.ready();
+      await localforage.clear();
+      return;
+    }
+
+    storageReference.clear();
+  }
+
+  static async getStorageRankings() {
+    const value = await Cache.get(STORAGE_RANKINGS_KEY, false);
+    if (value && typeof value === 'string') {
+      JSON.parse(value);
+    }
+    return {};
+  }
+
+  static async handleStorageRankingUpdate(key: string) {
+    const value = await Cache.getStorageRankings();
+    value[key] = value[key] ? parseInt(value[key], 10) + 1 : 1;
+    // tslint:disable-next-line:no-floating-promises
+    Cache.put(STORAGE_RANKINGS_KEY, value);
+  }
+}
+
+function overrideLocalForageSupport() {
+  // This fixes an existing bug in localForage which checks for fetch to exist to check if Safari version is later than 10.1
+  // Angular patches fetch in Safari and the test to check if fetch is native code returns false
+  // Once LocalForage fixes this bug, we can remove this method
+  // tslint:disable-next-line
+  localforage['originalSupports'] = localforage.supports;
+  localforage.supports = (driver: string) => {
+    try {
+      const isSafari = /(Safari|iPhone|iPad|iPod)/.test(navigator.userAgent) &&
+            !/Chrome/.test(navigator.userAgent) &&
+            !/BlackBerry/.test(navigator.platform);
+      if (isSafari) {
```

Comment:
there's another class in taurus that's checking browser name, it just doesn't check the version.